app/models/lookup_value.rb

```diff
@@ -37,6 +38,11 @@ def value=(val)
     end
   end
 
+  def hidden_value=(val)
+    lookup_key.hidden_value = val
+    lookup_key.save unless lookup_key.new_record?
```

Comment:
autosave caused problems in other places which is why this was added, I'll change it save the lookup key only before the lookup value is saved and only if `hidden_value` was updated.